python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -246,23 +246,18 @@ def _get_status_for_asset_key(
         ) -> Union[PartitionedAssetBackfillStatus, UnpartitionedAssetBackfillStatus]:
             if self.target_subset.asset_graph.get_partitions_def(asset_key) is not None:
                 materialized_subset = self.materialized_subset.get_partitions_subset(asset_key)
-                failed_partitions = set(
-                    self.failed_and_downstream_subset.get_partitions_subset(
-                        asset_key
-                    ).get_partition_keys()
-                )
-                requested_partitions = set(
-                    self.requested_subset.get_partitions_subset(asset_key).get_partition_keys()
-                )
+                failed_subset = self.failed_and_downstream_subset.get_partitions_subset(asset_key)
+                requested_subset = self.requested_subset.get_partitions_subset(asset_key)
 
                 return PartitionedAssetBackfillStatus(
                     asset_key,
                     len(self.target_subset.get_partitions_subset(asset_key)),
                     {
                         AssetBackfillStatus.MATERIALIZED: len(materialized_subset),
-                        AssetBackfillStatus.FAILED: len(failed_partitions),
-                        AssetBackfillStatus.IN_PROGRESS: len(requested_partitions) - (
-                            len(failed_partitions & requested_partitions) + len(materialized_subset)
+                        AssetBackfillStatus.FAILED: len(failed_subset - materialized_subset),
+                        AssetBackfillStatus.IN_PROGRESS: len(requested_subset) - (
```

Comment:
I'm finding this a little bit difficult to follow. Is there some name that would make sense to assign to `(failed_subset & requested_subset) - materialized_subset`?

Reply:
yup, I think calling this `failed_and_requested` would be good. I added a commit with some comments that should make this more readable.